src/components/tables/table_dropdown_button/table_dropdown_button.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, useState } from "@odoo/owl";
+import { ActionSpec } from "../../../actions/action";
+import { DEFAULT_TABLE_CONFIG } from "../../../helpers/table_presets";
+import { interactiveCreateTable } from "../../../helpers/ui/table_interactive";
+import { _t } from "../../../translation";
+import { SpreadsheetChildEnv } from "../../../types";
+import { TableConfig } from "../../../types/table";
+import { ActionButton } from "../../action_button/action_button";
+import { PopoverProps } from "../../popover/popover";
+import {
+  CustomTablePopoverMouseEvent,
+  TableStylesPopover,
+} from "../table_styles_popover/table_styles_popover";
+
+interface State {
+  popoverProps: PopoverProps | undefined;
+}
+
+export class TableDropdownButton extends Component<{}, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-TableDropdownButton";
+  static components = { TableStylesPopover, ActionButton };
+  static props = {};
+
+  state = useState<State>({ popoverProps: undefined });
+
+  onStylePicked(styleId: string) {
+    const sheetId = this.env.model.getters.getActiveSheetId();
+    const tableConfig = { ...this.tableConfig, styleId };
+    const result = interactiveCreateTable(this.env, sheetId, tableConfig);
+    if (result.isSuccessful) {
+      this.env.openSidePanel("TableSidePanel", {});
+    }
+    this.closePopover();
+  }
+
+  onClick(ev: CustomTablePopoverMouseEvent) {
+    if (ev.hasClosedTableStylesPopover || this.state.popoverProps) {
```

Comment:
please don't :'( Is it really necessary or could you maintain a state to toggle it?

Reply:
I think we have to find a better way globally (including for the existing similar case). I'd say we can copy that (shitty) way of managing this toggle and refactor it in a dedicated PR later (not sure how we should do it yet).